### DISCUSSION TITLE
3.0.0 is out!

### DISCUSSION THREAD

**Discussion by noah-nuebling** [noah-nuebling/mac-mouse-fix#743] [Other]
# Mac Mouse Fix 3 is here! ⭐️

Check out the <a href='https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/3.0.0'>3.0.0 release notes</a>.

If you guys have any questions or feedback let me know!

**Comment by vordenken** [comment]
Hey Noah,
first of all - thanks for your work on this absolutely amazing app. I use it daily for the last year and cannot live without it anymore.
But one question - what exactly are the change from the latest release of version 2 to the new version 3? 
I don't want to lose my current settings and workflows using Mac-mouse-fix.
Is there like a comparison between v2 and v3? And do I keep my settings?

**Reply by unickq** [reply]
The same question: is monetization the only change from the 2 to 3 version? 
I'm not saying paying $2 for software is wrong - no prob with this, but I just wanted to know because I couldn't find it in the release notes.

**Reply by noah-nuebling** [reply]
Hey guys, thanks for reaching out and for the nice words!

I'll address your questions and concerns below:

## What's new in MMF 3

There are many changes in MMF 3 compared to 2. Some of the most important are:

- **Click and Drag to Scroll & Navigate**

   You can now click and drag to simulate swiping with 2 fingers on an Apple trackpad. This lets you breeze through the system as it allows you to scroll really precisely and flexibly in 360°, go back and forward in Safari, delete mails quickly and much more.

   You can even use [Swish](https://highlyopinionated.co/swish/) right from your mouse. It's a really nice app that's designed to let you manage your windows with a simple swipe on your trackpad. But now, you can use it from your mouse, too! - Anything you can do by swiping on a trackpad works just as well with Mac Mouse Fix.

- **Click and Scroll Gestures**

  You can now hold a mouse button and then scroll to trigger many different actions, such as opening Launchpad, zooming in or out, and much more. It feels super responsive and fluid, just like a trackpad.

- **Totally rewritten the scrolling**

   Scrolling now requires less effort and feels much smoother and more fun to use. It can also simulate swiping on a trackpad. This brings some small niceties such as a little bounce when you reach the end of a page, being able to go back and forward in Safari by scrolling horizontally, better compatibility with Preview.app, and more.

- **Awesome new UI** 

  The apps UI is also completely rewritten to be more beautiful, powerful and intuitive. It can also be translated into other languages now and it's already available in Chinese, Korean, and German.

- **Menubar Item**

  There's now a menubar item that you can enable if you like which lets you see when Mac Mouse Fix is active, and it looks sleek.

- **Do everything you do on a Trackpad**

  If you're using a 3-button mouse, I think the biggest difference between MMF 2 and MMF 3 in your daily experience will be the scrolling. But if you use a 5-button mouse, I think MMF 3 is quite transformative, because it really lets you do anything you can do on a trackpad with simple and intuitive gestures on your mouse. For example, Click and Scroll to *Zoom in or out*, Click and Drag to go *Back and Forward* in Safari, and much more.

For an overview of the new trackpad-like gestures in MMF 3 (including Video Demos), see the [trackpad section](https://macmousefix.com/#trackpad) on the website. There's also a [scrolling section](https://macmousefix.com/#scroll), which shows the new, smoother and easier scrolling. But I think it's best to just try it out yourself.

To see a more detailed log of all the changes you can view the release notes of MMF [3.0.0 Beta 7](https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/3.0.0-Beta-7) and the previous Betas.

## How to Restore Your Settings After You Go Back to Mac Mouse Fix 2

If you update from MMF 2 to MMF 3, your current settings will be deleted. 
However, there's a way to **save** and **restore** the settings, after you go back to MMF 2:

1. Open the 'More...' screen in MMF 2, then click the MMF logo in the top left while holding Command (⌘) and Shift (⇧) on your keyboard. This will reveal a file called 'config.plist' in the Finder. This file contains all your settings.
2. Save a copy of 'config.plist', to restore later
3. Update to MMF 3 and use it as you wish
4. If you want to go back to MMF 2, just download it again, then reveal 'config.plist' in Finder and replace it with the copy you made earlier.\
   *If MMF 2 doesn't enable after going back to it, you might have to delete MMF 3, then empty the trash, and then restart your computer.*

---

I hope this helps you make a decision whether you want to check out MMF 3, and gives you peace of mind about going back to MMF 2 in case you'd like to!

Let me know if you have more thoughts or questions to share.

**Comment by Iron-Ham** [comment]
Hi @noah-nuebling  –

I'm happy to pay for Mac Mouse Fix, as I've gotten a ton of value out of it. Just a quick question (+ one possible issue  related to the trial period):

- I use two macOS devices, and I'm the crazy kind of person who gets a new Mac every few years. How many devices is the new license approved for, and how does device de-activation work (for those of us that swap machines). 
- I'm getting a notification that my trial period is over. However: I've used my laptop once since December 19, and the release hasn't been out for 30 days. 

Anyhow, I'm happy to pay a full $10 for this. Thank you!

**Reply by noah-nuebling** [reply]
Hey there @Iron-Ham, one license should work for all of your current and future devices. It’s designed so you don’t have to worry about de-activation. If there are any problems, just contact me.

Thanks for informing me about the bugs you’re experiencing with the trial period. I didn’t expect users to frequently switch machines during the trial period, and there’s probably a bug in the logic around that. 
If you want a free license, just shoot me an email.

I hope that helps and I’m glad you like the app!
Let me know if you have more questions.

**Reply by Iron-Ham** [reply]
Thanks for getting back to me so quickly. I've gone ahead and picked up the $10 support option. 

Right now I use a combination of the Logitech G-Hub and Mac Mouse Fix in order to get the config I need. My mouse has a ton of buttons (11, I think?) and my understanding is that the only really supported buttons on an OS-level are Mouse 4 and Mouse 5 – which means that I have to assign some of these using G-Hub (which is unfortunate; a lot of those bindings are _worse_ – notably the lack of a trackpad simulated back/forward).

**Reply by noah-nuebling** [reply]
At an OS level, Mac Mouse Fix supports all Mouse buttons up to button 32. However many Logitech mice don’t send standard button events when you click their buttons. Instead they send proprietary signals that are designed to work with their special driver software like G Hub or Options. It might be possible to get the buttons to work with Mac Mouse Fix by somehow reprogramming the mouse to send standard button events in G Hub and then uninstalling G-Hub. I don’t know the specifics of this, though, so I can’t give more detailed advice at the time.

I hope this helps a little, and thank you very much for the milkshakes! :)

**Reply by Iron-Ham** [reply]
@noah-nuebling Appreciated! Do you know of any multi-button mouse that sends standard mouse button events when you click their buttons? I'll hack the Logitech if I have to, but it'd be easier to pick up a new mouse instead 😅

**Reply by noah-nuebling** [reply]
Hey @Iron-Ham, I don’t have personal experience with multi-button mice so I can’t tell for sure. I always made good experiences with these unknown Asian brands with good ratings from Amazon. Like [this one](https://www.amazon.de/Trust-Gaming-Individuell-Seitenplatte-RGB-Beleuchtung/dp/B08C7TZYJZ/ref=mp_s_a_1_7?crid=1X2LT1GJ7S7AV&keywords=mouse+multi+button&qid=1704827353&sprefix=mouse+multi+button%2Caps%2C100&sr=8-7) looks pretty cool I think. However, all the multi-button mice I could find with a quick search are designed to be used with proprietary driver software, which makes it thinkable that the buttons also don’t send standard button events and don’t work with Mac Mouse Fix. 

I would assume that the buttons send standard events out of the box because that would make the most sense to me by far, but the Logitech mice don’t do that either for some reason, so I can’t tell you for sure.

So unfortunately I can’t give you a great answer but I hope it’s still a little bit helpful. Let me know if you have more questions.

Edit: Linked the wrong mouse

**Reply by Iron-Ham** [reply]
Certainly an unorthodox solution on my end: deleting the Logitech firmware, using Steermouse to map the buttons to standard mouse buttons, and then using MacMouseFix to handle those events.

Some preliminary searching tells me that Logitech doesn't support mouse buttons beyond MB5 because Windows only supports up to five mouse buttons. Their primary business – especially for the G502x (and similar mice) – seems to be centered around gaming, which generally implies Windows.

**Reply by spacecakes** [reply]
I'm curious though how macOS detects my MX Master 3's thumb button clicks without any Logitech software installed. It does _something_ with it, not just something very useful.

**Reply by noah-nuebling** [reply]
@spacecakes I have an MX Master 2s and I think when you press the Gesture button, it sends alt-tab keyboard shortcut signals by default. Which is of course not very useful under macOS and which prevents the button from working with Mac Mouse Fix.

**Reply by spacecakes** [reply]
Yeah that makes sense. Well I mean, it really doesn't, but it's consistent with the behavior I'm seeing. 😅 Would be great if it did work with MMF.

**Reply by noah-nuebling** [reply]
@spacecakes It would really be great to add support! But I don't think I'll adress this very soon. I want to make MMF the best experience for standard mice first and then later try to focus on supporting all those mice that send special signals. At least that's the plan for now. In the mean time, you might enjoy using [SteerMouse](https://plentycom.jp/en/steermouse/) with your MX Master, since it has really good device support.

**Reply by spacecakes** [reply]
@noah-nuebling perfectly understandable! I did use SteerMouse before I found MMF. I can't recall exactly why I switched but I think it was because MMF feels much more modern and intuitive. So I decided to live without the thumb button, which is pretty awkwardly placed and mushy to press anyway.

**Reply by noah-nuebling** [reply]
> "MMF feels much more modern and intuitive" 

That's really motivating to hear, thanks for sharing! 

> I decided to live without the thumb button, which is pretty awkwardly placed and mushy to press anyway

Yeah I really don't like that button either! Finally someone agrees haha. 😎🤜🤛😎

**Comment by reorx** [comment]
I saw the notification for payment and I paid. I hope to see more indie apps that I use everyday adopt this kind of charming monetization.

**Comment by jaknz** [comment]
I've just discovered this software while searching for an alternative to Logi's awful awful software. I'm more than happy to pay for it too! This is what Logi should have made in the first place, not the AI-encumbered phone-home bloatware they're shipping, including zero-day vulnerabilities on open unauthenticated ports. What a mess. Thank you for this gorgeous alternative!!! ❤️

**Comment by sk-izsk** [comment]
Hello, interested to buy your app. Wondering if i buy version 3 will i get all future versions update like v4 v5 etc.? Thanks

**Reply by noah-nuebling** [reply]
Hello @sk-izsk thank you the question! I've not decided on this yet.

I plan to perhaps do paid upgrades for major version changes like v4 and v5. But I know this wouldn't be the optimal user experience so I plan to avoid it if it's financially feasible.

Either way, I think it will be quite a while until a v4 release, since almost everything I've been working on and that I plan to do is not new features but improving fundamental stuff like the localization system, update system, licensing system, bug report system, tweaking the scrolling algorithms, a few more options in the UI, etc.

It's all about improving fundamental things to make existing functionality better and lay groundwork for the future. None of this stuff would warrant a v4 release in my eyes.

Well, the only thing that I've put significant work into which might warrant a v4 release is pointer-speed-settings. I've built some great stuff there before the 3.0.0 release but I cut it because I thought it took too much time and the other features were already a great package.

But I'm not focused on this now, and I think it's very unlikely this will be coming soon. I plan to work on a lot of other things first, before tackling this, and even then it would take several months to implement I believe.

Based on this - I can't look into the future, so I don't want to make promises - but I think, realistically, it will take *at least a year* from now until a v4 release. And, as long as the app is still financially sustainable at that point, it will not be a paid update.

Does this sound ok to you, or would you feel bad buying the app under the prospect of perhaps having to pay again in 1 or 2 years to continue getting major updates? Your perspective would be valuable. Thank you.

I hope this helps a bit. And let me know if you have any more questions.

**Reply by spacecakes** [reply]
@noah-nuebling Please **do** charge for future major versions, either with upgrade pricing or with a (reasonable) subscription model.

**Reply by sk-izsk** [reply]
Hello @noah-nuebling  , Thanks for taking your time to reply me , I really appriciate that.

Typically I dont buy those software where i need to keep buying for major version update. But the way you take your time to entertain my concern , even though at that time i havent bought it and still you explain all this with such humbleness, I am really impressed and appriciated .

I see how many times you edited the message to explain me better way , that shows your dedication for your users. and everywhere you keep reply with such humbleness . That impress me a lot.

Also you keep the price such affordable that majority can buy it , it shows your care for your users.

Also I see you make MMF2 free completely for those who cannot pay for software , moreover you keep updating that software so that they can use it in their latest os. This shows your empathy . Also I see you distribute free license to few people.

Although its your product, your policy and price and yet you bother to ask me if I am feeling bad or not . Thank you for this humbleness. 

I dont know if I am going to use this software or not (I havent even tried it yet) but I bought it, since I am a software engineer so majority of my work is keyboard based and i am just used to of using keyboard but still it just a small token from my end to show my appreciation and telling you a big thank you. 

I will try to buy again if you decide to price for major version and I have trust if you go for that you make the price that majority can afford. 

Keep up the good work, and most importantly you have great attitude, a down to earth behavior, humbleness and such caring character you have that is gem in todays society and we need more people like you . and its a request try not to change who you are  

Wish you best of luck . God bless you  my friend .  🙂

**Reply by Iron-Ham** [reply]
@noah-nuebling I'd happily pay for a major upgrade. This app -- in combination with SteerMouse to override the button events that my Logitech mouse sends out -- are core to my workflow.

I just wish I didn't have to override my G502x (and relatively often!) but it is what it is.

**Reply by noah-nuebling** [reply]
Hello guys,

Thanks for the positive feedback! I'm glad my explanation was appreciated and that you guys feel so positively about supporting the app.

@sk-izsk, thank you for your *super kind* message! While I do feel a bit uncomfortable with personal praise, I'm really grateful that you took the time to write such nice words. I do care about making the app a great experience that is accessible to everyone. So if the price is a barrier, please feel free to [send me an email](https://redirect.macmousefix.com/?target=mailto-noah), and we'll figure something out.

@spacecakes, thank you for your feedback on pricing.
I personally don't like the idea of a subscription model for individual users (I wouldn't want to use that kind of app myself). I was thinking about perhaps introducing some form of subscription-based, multi-seat license for businesses, but not for individuals. I do appreciate your generosity and thoughts on this.

@Iron-Ham Thanks for sharing your thoughts and experience. I hear you on the Logitech issues and I do plan to address this, but it will probably take a while. Currently, my priority is to polish the app's core functionality and create the best possible experience for standard mice. Once those foundations set are in place, I plan to build high-quality support for specific mice, like the ones from Logitech. I know how important Logitech support is for many users, but I think this strategy is best for the long-term health and quality of the app.

Thanks again for your feedback and support! It's very motivating and it helps me make better decisions.

**Reply by spacecakes** [reply]
@noah-nuebling subscription definitely isn't my first choice either, especially not for set-and-forget style of utilities like this, but it's also important it's a working business modell to sustain development. Worse than a subscription is a utility depend on daily become abandonware IMO. That can happen anyway and for any reason of course, but I do think having some monetization (and therefore developer incentive) makes it less likely.

**Reply by noah-nuebling** [reply]
@spacecakes you're right that I need some form of income to be able to sustain this, but it's currently working pretty well and I have several things I can try to increase revenue if necessary before even thinking about a subscription model, like increasing the price a little bit. So I'm really glad the app would be worth it to you! But a subscription is definitely not on the table anytime soon, and I don't think it ever will be.

**Reply by Iron-Ham** [reply]
> I personally don't like the idea of a subscription model for individual users (I wouldn't want to use that kind of app myself). I was thinking about perhaps introducing some form of subscription-based, multi-seat license for businesses, but not for individuals. I do appreciate your generosity and thoughts on this.

I think this makes sense. A utility that I feel successfully blurs this line is ScreenStudio, where a purchase is a perpetual use license that snags you 2 years of updates. Further updates require a new purchase. 


> @Iron-Ham Thanks for sharing your thoughts and experience. I hear you on the Logitech issues and I do plan to address this, but it will probably take a while. Currently, my priority is to polish the app's core functionality and create the best possible experience for standard mice. Once those foundations set are in place, I plan to build high-quality support for specific mice, like the ones from Logitech. I know how important Logitech support is for many users, but I think this strategy is best for the long-term health and quality of the app.

I wouldn't even sweat Logitech support until down the line. Though if it worked well I'd happily pay extra for it as an added feature. Others may feel differently but my take boils down to:

- you've built a generically useful utility. 
- support for niche products or nonstandard behaviors warrants niche pricing.
